src/driver.rs

```diff
@@ -366,3 +352,140 @@ impl MajorObject for Driver {
 }
 
 impl Metadata for Driver {}
+
+/// A wrapper around `GDALDriverManager`.
+/// This struct helps listing and registering [`Driver`]s.
+pub struct DriverManager;
+
+impl DriverManager {
+    /// Returns the number of registered drivers.
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// println!("{} drivers are registered", DriverManager::count());
+    /// ```
+    /// ```text
+    /// 203 drivers are registered
+    /// ```
+    pub fn count() -> usize {
+        _register_drivers();
+        let count = unsafe { gdal_sys::GDALGetDriverCount() };
+        count
+            .try_into()
+            .expect("The returned count should be zero or positive")
+    }
+
+    /// Returns the driver with the given index, which must be less than the value returned by
+    /// `DriverManager::count()`.
+    ///
+    /// See also: [`count`](Self::count)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// assert!(DriverManager::count() > 0);
+    /// let d = DriverManager::get_driver(0)?;
+    /// println!("'{}' is '{}'", d.short_name(), d.long_name());
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// 'VRT' is 'Virtual Raster'
+    /// ```
+    pub fn get_driver(index: usize) -> Result<Driver> {
+        _register_drivers();
+        let c_driver = unsafe { gdal_sys::GDALGetDriver(index.try_into().unwrap()) };
+        if c_driver.is_null() {
```

Comment:
I don't think `GDALGetDriver` sets the last error: https://github.com/OSGeo/gdal/blob/master/gcore/gdal_priv.h#L1734.

Reply:
[141c8d8](https://github.com/georust/gdal/pull/324/commits/141c8d8f4fab47f28a00010c18dff135bc2fb56e)